samples/ck-rx65n/basic-sample/src/iotconnect_app.c

```diff
@@ -167,6 +178,63 @@ static void publish_telemetry() {
     }
 }
 
+
+#ifdef SYMMETRIC_KEY_INPUT
+
+#define CREDENTIALS_BLOCK_DATA FLASH_DF_BLOCK_24
+
+
+// be careful with changing these constants
+#define IOTC_CONFIG_BUFF_LEN_GENERIC 64
+#define IOTC_CONFIG_BUFF_LEN_SYMMKEY 256
+typedef struct credentials {
+	char cpid[IOTC_CONFIG_BUFF_LEN_GENERIC];
+	char env[IOTC_CONFIG_BUFF_LEN_GENERIC];
+	char duid[IOTC_CONFIG_BUFF_LEN_GENERIC];
+	char symmkey[IOTC_CONFIG_BUFF_LEN_SYMMKEY];
+} credentials_t;
+
+static flash_err_t save_creds_to_file(void *data, uint32_t address, size_t size){
+	
+    flash_err_t ret;
+
+    ret = R_FLASH_Erase(address, size);
+    if (ret != FLASH_SUCCESS)
+    {
+        printf("Failed to erase credentials data before writing\r\n");
+        return ret;
+    }
+
+    ret = R_FLASH_Write((uint32_t*)data, address,
+                size);
+
+    if (ret != FLASH_SUCCESS) {
+        printf("Failed to write credentials data\r\n");
+    }
+
+    return ret;
+}
+    
+
+static flash_err_t load_creds_from_file(void **data, uint32_t address){
```

Comment:
void ** -> credentials_t ** ??

If this is just for credentials -- could hardwire address in the routine?